desc/geometry/curve.py

```diff
@@ -817,8 +827,21 @@ def __init__(
             errorif(knots[-1] > 2 * np.pi, ValueError, "knots must lie in [0, 2pi]")
             knots = knots[:-1] if closed_flag else knots
 
+        if break_indices is None:
+            unbroken_spline_intervals = [[]]
+        else:
+            # check that input is monotonic
+            assert np.all(
```

Comment:
minor point: if break_indices contains duplicates than np.unique will be a different length than the original, so the `==` won't broadcast and you won't ever get to the assert, and it might give an uninformative error message. Also, np.unique already sorts the values so you don't need sorted as well.

Might be better as
```python
unique = np.unique(break_indices)
errorif(len(unique) != len(break_indices), ValueError, "break_indices must not contain any duplicated")
errorif(np.any(unique != break_indices), ValueError, "break_indices must be monotonic")
```

Reply:
I realized we could also modify the error above this that I didn't see before

```
errorif(
        not np.all(np.diff(sorted(break_indices)) > 0),
        ValueError,
        "supplied break_indices must be monotonically increasing",
)
```
but maybe it's better to separate them like you did so errors are more clear

Reply:
I made them separate